internal/cloudns/resource_dns_record.go

```diff
@@ -96,13 +104,13 @@ func resourceDnsRecordRead(ctx context.Context, d *schema.ResourceData, meta int
 		if wantedId == actualId {
 			err = updateState(d, &zoneRecord)
 			if err != nil {
+				tflog.Error(ctx, err.Error())
 				return diag.FromErr(err)
 			}
 			return nil
 		}
 	}
 
-	d.SetId("")
```

Comment:
From https://learn.hashicorp.com/tutorials/terraform/provider-setup?in=terraform/providers I assumed this was necessary to handle manually deleted resources, otherwise they can't get out of the state file? 🤔 

(This section specifically)

![2022-04-22 11_34_26-Setup and Implement Read _ Terraform - HashiCorp Learn and 1 more page - Persona](https://user-images.githubusercontent.com/875533/164697392-f65e3229-fdcb-402e-bc1b-ca4645c1033d.png)

Though I might have misunderstood it?